server/src/com/cloud/network/vpc/VpcManagerImpl.java

```diff
@@ -747,6 +810,67 @@ public VpcOffering updateVpcOffering(final long vpcOffId, final String vpcOfferi
     }
 
     @Override
+    public CIDR[] getZoneSuperCidrList(final long zoneId) throws BadCIDRException {
+        final String superCIDRList = _dcDao.getDetail(zoneId, Params.SUPER_CIDR.name());
+        if (superCIDRList != null) {
+            String[] str_cidr_list = superCIDRList.split(",");
+            CIDR[] cidr_list = NetUtils.convertToCIDR(str_cidr_list);
+            if (!NetUtils.cidrListConsistency(cidr_list)) {
+                throw new InvalidParameterValueException("The cidr list is not consistent " + Arrays.toString(cidr_list));
+            }
+            return cidr_list;
+        } else {
+            throw new InvalidParameterValueException("Zone super cidr is null check cloud.data_center_details for zone id=" + zoneId);
+        }
+    }
+
+    private List<CIDR> getAllVpcCidrs() throws BadCIDRException {
+        // get all the used cidrs
+        List<CIDR> usedSubnets = new ArrayList<CIDR>();
+        for (Vpc vpc : _vpcDao.listAll()) {
+            if (_vpcSrvcDao.areServicesSupportedInVpc(vpc.getId(), Service.VPCDynamicRouting)) {
```

Comment:
For performance reasons, calling a query in a loop should be avoided.  Is it possible to refactor this ``for`` loop into a join?